### PR TITLE
Porting parent_run_id fix #1137 into branch-0.9.1

### PR DESCRIPTION
**Pull request by pogil**
## What changes are proposed in this pull request?
 
Cherry-picking #1137 for parent-id backward compatibility: Reintroduces the `parent_run_id` argument to `MlflowClient.create_run`. In https://github.com/mlflow/mlflow/pull/978 we removed this argument, which broke some third-party MLflow integrations. We ultimately do want to make the change in #978 with MLflow 1.0 (i.e., remove the `parent_run_id` argument) - this PR just aims to defer making that change till the 1.0 release.
 
Also updates tensorflow dependency in tests.

## How is this patch tested?
 
Existing unit tests verify that codepaths that call `create_run` and pass parent run ID as a tag still work
Added new unit tests verifying that passing parent_run_id explicitly to `create_run` works.
 
### Is this a user-facing change? 

- [ ] No. Add the `rn/none` label, then you can skip the rest of this section.
- [X] Yes. Give a description of this change to be included in the release notes for MLflow users.
 
[Python API] Reintroduces the parent_run_id argument to MlflowClient.create_run() to avoid making a breaking change to the create_run API. We will ultimately make this breaking change & remove the parent_run_id argument in MLflow 1.0.
 
### What component(s) does this PR affect?
 
- [ ] UI
- [ ] CLI 
- [ ] API 
- [ ] REST-API 
- [ ] Examples 
- [ ] Docs
- [X] Tracking
- [ ] Projects 
- [ ] Artifacts 
- [ ] Models 
- [ ] Scoring 
- [ ] Serving
- [ ] R
- [ ] Java
- [X] Python

### Please add one label to the PR so it can be classified correctly in the release notes. Options:
 
* `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
* `rn/none` - No description will be included. The PR will be mentioned just by the PR number in the "Small Bugfixes and Documentation Updates" section
* `rn/feature` - A new user-facing feature worth mentioning in the release notes
* `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
* `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: test-requirements.txt [hunk 1]
```diff
@@ -17,7 +17,7 @@ pytest-cov==2.6.0
 rstcheck==3.2
 scikit-learn==0.20.2
 scipy==1.2.1
-tensorflow==1.12.0
+tensorflow==1.13.1
```
Comment (mparkhe) [inline test-requirements.txt]: Which tests need this upgrade?

**Review by mparkhe — APPROVED** [review]
One question about the TF library upgrade. LGTM otherwise...

File: test-requirements.txt (hunk 1 above)
Comment (pogil) [reply test-requirements.txt]: I couldn't run `pip install -r test-requirements.txt` without this change.

Comment (smurching) [reply test-requirements.txt]: ^I think this issue (needing to bump the TF version) would have been addressed by this PR btw, which updates our contributor guide to standardize dependencies/Python versions/tests run locally & on Travis: https://github.com/mlflow/mlflow/pull/1097. Specifically I think the issue is that TF 1.12.0 isn't available in certain Python versions (like 3.7)

Comment (smurching) [reply test-requirements.txt]: Good to bump the dependency version anyways to make sure we're testing against the latest TF

**Pull request merged**